### PR TITLE
Update region list for AWS AMI publishing

### LINKED ISSUE #6277: Teleport AMI for AWS region eu-north-1
### What
Please publish your amis in the Stockholm region. They're currently not available and copying returns a permission error.

### Why
I run my aws infra(both personal and work) mainly on eu-north-1 and it would be nice to have the official amis available for use in the same region.

### PR DESCRIPTION
**Pull request by webvictim**
Fixes #6277 

Adds new AWS regions that are available to the public (i.e. non-opt-in)

Also removes unused `S3_BUCKET_ID` variable.

### CODE REVIEW

File: assets/aws/Makefile [hunk 1]
```diff
@@ -23,17 +23,14 @@ TELEPORT_UID ?= 1007
...
+# For now, limit AMI publishing to non opt-in regions
+# https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html
+# You can get this with $(AWS_DEFAULT_REGION=us-west-2 aws ec2 describe-regions | jq -r '.Regions | map(.RegionName) | join(",")')
+DESTINATION_REGIONS ?= eu-north-1,ap-south-1,eu-west-3,eu-west-2,eu-west-1,ap-northeast-3,ap-northeast-2,ap-northeast-1,sa-east-1,ca-central-1,ap-southeast-1,ap-southeast-2,eu-central-1,us-east-1,us-east-2,us-west-1,us-west-2
 
 # Generate timestamp for builds
 BUILD_TIMESTAMP := $(shell TZ=UTC /bin/date "+%Y%m%d-%H%M%S%Z")
 
-# S3 Bucket ID
-S3_BUCKET_ID ?=
```
Comment (awly) [inline assets/aws/Makefile:35]: remove the mention of `S3_BUCKET_ID` from https://github.com/gravitational/teleport/blob/master/assets/aws/README.md#L44

File: assets/aws/Makefile [hunk 2]
```diff
@@ -23,17 +23,14 @@ TELEPORT_UID ?= 1007
 INSTANCE_TYPE ?= t2.medium
 
 # Use comma-separated values without spaces for multiple regions
-# For now, limit AMI to regions with DynamoDB encryption at rest enabled
```
Comment (awly) [inline assets/aws/Makefile:26]: was there a specific reason we limited the regions this way before?

Comment (webvictim) [reply assets/aws/Makefile:26]: Back when DynamoDB encryption at rest was first launched, it was only supported in specific regions. It's now supported everywhere - see the bottom of https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/EncryptionAtRest.html

**Review by awly — APPROVED** [review]

**Review by russjones — APPROVED** [review]
Bot.

**Pull request merged**